packages/vm/core/testcore/base_test.go

```diff
@@ -60,18 +58,68 @@ func TestLedgerBaseConsistency(t *testing.T) {
 
 // TestLedgerBaseConsistencyWithRequiredTopUpFee deploys a chain and checks the consistency of L1 and L2 ledgers after topping up fees
 func TestLedgerBaseConsistencyWithRequiredTopUpFee(t *testing.T) {
-	env := solo.New(t)
-	ch, _ := env.NewChainExt(nil, isc.TopUpFeeMin/2, "chain1", evm.DefaultChainID, governance.DefaultBlockKeepAmount)
+	env := solo.New(t, &solo.InitOptions{
+		Debug:           true,
+		PrintStackTrace: true,
+	})
 
+	ch, _ := env.NewChainExt(nil, isc.GasCoinTargetValue/2, "chain1", evm.DefaultChainID, governance.DefaultBlockKeepAmount)
 	ch.CheckChain()
 
-	l2Total1 := ch.L2TotalAssets().BaseTokens()
-	someUserWallet, _ := env.NewKeyPairWithFunds()
-	ch.DepositBaseTokensToL2(1*isc.Million, someUserWallet)
-	l2Total2 := ch.L2TotalAssets().BaseTokens()
-	require.Equal(t, l2Total1+1*isc.Million, l2Total2)
+	require.EqualValues(
+		t,
+		isc.GasCoinTargetValue/2,
+		ch.L2BaseTokens(accounts.CommonAccount()),
+	)
+	require.EqualValues(
+		t,
+		0,
+		ch.L2BaseTokens(ch.OriginatorAgentID),
+	)
+	require.EqualValues(t, coin.Value(isc.GasCoinTargetValue/2), ch.L2TotalBaseTokens())
+
+	gasCoinValueBefore := coin.Value(ch.GetLatestGasCoin().Value)
 
+	someUserWallet, someUserAddr := env.NewKeyPairWithFunds()
+	_, _, vmRes, ptbRes, err := ch.PostRequestSyncTx(
+		solo.NewCallParams(accounts.FuncDeposit.Message()).
+			AddBaseTokens(isc.GasCoinTargetValue*4).
+			WithGasBudget(math.MaxUint64),
+		someUserWallet,
+	)
+	t.Logf("PTB gas fee: %d", ptbRes.Effects.Data.GasFee())
+	require.NoError(t, err)
 	ch.CheckChain()
+
+	gasCoinValueAfter := coin.Value(ch.GetLatestGasCoin().Value)
+	t.Logf("gasCoinValueBefore: %d, gasCoinValueAfter: %d", gasCoinValueBefore, gasCoinValueAfter)
+	require.EqualValues(
+		t,
+		isc.GasCoinTargetValue-coin.Value(ptbRes.Effects.Data.GasFee()),
+		gasCoinValueAfter,
+	)
+
+	totalDeposited := coin.Value(isc.GasCoinTargetValue/2 + isc.GasCoinTargetValue*4)
+	totalDeductedForGasCoin := isc.GasCoinTargetValue - gasCoinValueBefore
+	totalL2Tokens := totalDeposited - totalDeductedForGasCoin
+	require.EqualValues(t, coin.Value(totalL2Tokens), ch.L2TotalBaseTokens())
+
+	addedToCommonAccount := min(isc.GasCoinTargetValue, vmRes.Receipt.GasFeeCharged)
```

Comment:
This line feels strange. Could you pls add comment why is it so?
Because for me it feels like it should be like:
```suggestion
	addedToCommonAccount := min(isc.GasCoinTargetValue - gasCoinValueBefore, vmRes.Receipt.GasFeeCharged)
```
Like, why would we deduct from fee and put on common account the entire `isc.GasCoinTargetValue` if it was not fully used?
Damn, its hard to think about this stuff in this late time